### ISSUE TITLE
Better message in case of not using parametrized variable

### ISSUE THREAD

**Issue by pwilczynskiclearcode** [pytest-dev/pytest#1539]
py.test 2.9.1

In case of not using one of parametrized variables the following message is displayed:
```
/python2.7/site-packages/_pytest/python.py:237: in pytest_generate_tests
    metafunc.parametrize(*marker.args, **marker.kwargs)
/python2.7/site-packages/_pytest/python.py:1000: in parametrize
    raise ValueError("%r uses no fixture %r" %(self.function, arg))
E   ValueError: <function test_abc at 0x7f9bb394dde8> uses no fixture u'suffix'
```
where `suffix` isn't a fixture but missed variable from `pytest.mark.parametrize`.

Suggested enhancement: Raise Exception with a message suggesting using the `suffix` variable and not confuse developers about `suffix` fixture.

**Comment by The-Compiler** [comment]
It *is* a fixture from pytest's point of view - but I'd also be okay with the error saying "argument" instead of "fixture". "variable" is even more confusing, IMHO.

**Comment by RonnyPfannschmidt** [comment]
for parametrize, in case of direct "arguments" we should use names that make the most sense

for a good error we should take a look if there is a indirect fixture that takes parameters (needs indirect=True) or if the parameter is just missing (maybe typo detection)

**Comment by RibeiroAna** [comment]
I'll work on this :)

**Comment by hackebrot** [comment]
I'd say we change the ``ValueError`` to report an issue

- with ``argument`` if parameter **is not** indirect
- with ``fixture`` if parameter **is** indirect

**Comment by hackebrot** [comment]
Please review #1757

**Comment by nicoddemus** [comment]
Fixed in #1757
